File: tournaments/countSumOfTwoRepresentations2/countSumOfTwoRepresentations2.cpp
int countSumOfTwoRepresentations2(int n, int l, int r) {
  int result = 0;

  for (int a = l; a <= r; a++) {
    int b = n - a;
    if (a <= b && b <= r) {
      result++;
    }
  }

  return result;
}

File: tournaments/countSumOfTwoRepresentations2/countSumOfTwoRepresentations2.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub countSumOfTwoRepresentations2 {
    my ($n, $l, $r) = @_;

    my $result = 0;

    while ($l <= $r) {
        if ($n - $l <= $r && $l << 1 <= $n) {
            ++$result;
        }
        ++$l;
    }

    return $result;
}
